jdaviz/configs/mosviz/helper.py

```diff
@@ -415,19 +416,18 @@ def load_data(self, spectra_1d=None, spectra_2d=None, images=None,
         instrument = kwargs.pop('instrument', None)
         msg = ""
 
-        if directory is not None and instrument is not None:
+        if directory is not None and Path(directory).is_dir():
+            if instrument not in ('nirspec', 'niriss'):
+                msg = ("Warning: Unspecific or Unrecognized MOS Instrument. Only JWST NIRSpec and "
+                       "NIRISS folder parsing is fully supported. Falling back to NIRSpec parsing")
+                warnings.warn(msg)
+                instrument = "nirspec"
             if instrument.lower() == "nirspec":
                 super().load_data(directory, "mosviz-nirspec-directory-parser")
             elif instrument.lower() == "niriss":
                 self.load_niriss_data(directory)
-            else:
```

Comment:
The if section above:

https://github.com/spacetelescope/jdaviz/blob/eced5e9c889f24198eff8857a7101a37312193f2/jdaviz/configs/mosviz/helper.py#L420-L424

should catch all instances where the user supplies something other than NIRISS or NIRSpec as an instrument keyword, hence an unneeded else, right? Though I suppose it could be rewritten AS an else statement if it doesn't match either case, THEN fallback? Though this way it's nice because it overrides the instrument keyword, and an unrecognized instrument follows the EXACT codepath as the NIRSpec case

Reply:
Ah, okay. I missed that. Ignore my comment on `else` then.